Add `odo logs` (#5760)
redhat-developer/odo: docs/website/versioned_docs/version-3.0.0/command-reference/logs.md

    endpoints:
    - name: http-3000
      targetPort: 3000
    image: registry.access.redhat.com/ubi8/nodejs-14:latest
    memoryLimit: 1024Mi
    mountSources: true
  name: runtime
- name: infinitepodone
  kubernetes:  
    inlined: |
      apiVersion: v1
      kind: Pod
      metadata:
        name: infinitepodone
      spec:
        containers:
          - name: main
            image: docker.io/dharmit/infiniteloop
- name: infinitedeployment
  kubernetes:
    inlined: |
      apiVersion: apps/v1
      kind: Deployment
      metadata:
        name: infinitedeployment
      spec:
        replicas: 1
        selector:
          matchLabels:
            app: infinite
        template:
          metadata:
            labels:
              app: infinite
          spec:
            containers:
            - name: main
              image: docker.io/dharmit/infiniteloop
 ```
When you do `odo dev`, odo creates pods for:
1. The component named `node` itself. Containers for this are created using `.components.container`.
2. Kubernetes component named `infinitepodone`
3. Kubernetes component named `infinitedeployment`. As can be seen under `.spec.template.spec.containers` for this 

>    particular component, it creates two containers for it.

Reviewer: ```suggestion
   particular component, it creates one container for it.
```

When you run `odo logs`, you should see logs from all these containers. Each line is prefixed with 
`<container-name>:` to easily distinguish which the container the logs belong to. Since we named multiple 

> containers in the `devfile.yaml` as `container`, `odo logs` has distinguished these containers as `container`, 

Reviewer: ```suggestion
containers in the `devfile.yaml` as `main`, `odo logs` has distinguished these containers as `main`
```

> `container1` and `container2`:

Reviewer: ```suggestion
and `main1`:
```

    component: runtime
    group:
      isDefault: true
      kind: debug
    workingDir: ${PROJECT_SOURCE}
  id: debug
- exec:
    commandLine: npm test
    component: runtime
    group:
      isDefault: true
      kind: test
    workingDir: ${PROJECT_SOURCE}
  id: test
components:
- container:
    endpoints:
    - name: http-3000
      targetPort: 3000
    image: registry.access.redhat.com/ubi8/nodejs-14:latest
    memoryLimit: 1024Mi
    mountSources: true
  name: runtime
- name: infinitepodone
  kubernetes:  
    inlined: |
      apiVersion: v1
      kind: Pod
      metadata:
        name: infinitepodone
      spec:
        containers:
          - name: container
            image: docker.io/dharmit/infiniteloop
- name: infinitedeployment
  kubernetes:
    inlined: |
      apiVersion: apps/v1
      kind: Deployment
      metadata:
        name: infinitedeployment
      spec:
        replicas: 1
        selector:
          matchLabels:
            app: infinite
        template:
          metadata:
            labels:
              app: infinite
          spec:
            containers:
            - name: container
              image: docker.io/dharmit/infiniteloop

>             - name: container

Author: Thanks for the catch. It was a wrong example. I have fixed it and named the containers `main`.

---
title: odo logs
---

## Description

> 

Author: What's astray here? I checked `dev.md`, `list.md` and `list-namespaces.md`. I don't follow what you're trying to say.

> 

Reviewer 2: Yes that! My GoLand does soft/hard wraps and it annoys me. I assumed that's what happened with you but looks like it was intended.

> 

Author: It wasn't intended, but in the past I used to hard wrap at 80 chars when I was using vim. So it went unnoticed to me that GoLand was doing the same.

  kubernetes:
    inlined: |
      apiVersion: apps/v1
      kind: Deployment
      metadata:
        name: infinitedeployment
      spec:
        replicas: 1
        selector:
          matchLabels:
            app: infinite
        template:
          metadata:
            labels:
              app: infinite
          spec:
            containers:
            - name: container
              image: docker.io/dharmit/infiniteloop
            - name: container
              image: docker.io/dharmit/infiniteloop

```
When you do `odo dev`, odo creates pods for:
1. The component named `node` itself. Containers for this are created using 
   `.components.container`.
2. Kubernetes component named `infinitepodone`
3. Kubernetes component named `infinitedeployment`. As can be seen under `.
   spec.template.spec.containers` for this particular component, it creates 
   two containers for it.

When you do `odo logs`, you should logs from all these containers. Each line 
is prefixed with `<container-name>:` to easily distinguish which the 
container the logs belong to. Since we named multiple containers in the 
`devfile.yaml` as `container`, `odo logs` has distinguished these containers 

> as `container`, `container1` and `container2`:

Reviewer: To disambiguate a little bit more, you could use container[1] instead

> as `container`, `container1` and `container2`:

Author: Makes sense. 97b28d3.

---
title: odo logs
---

> `odo logs` displays the logs for all the containers odo created for the 

Author: Thanks for pointing. Thanks a lot for creating a common pattern/template. 👍🏾 
I have made the changes, but not added 'Advanced Usage' since I couldn't think of putting anything in there. PTAL.
